lib/routes/account.js

```diff
@@ -484,13 +501,17 @@ module.exports = function (
                   uid: emailRecord.uid,
                   kA: emailRecord.kA,
                   wrapKb: wrapKb,
-                  emailVerified: emailRecord.emailVerified
+                  emailVerified: emailRecord.emailVerified,
+                  tokenVerificationId: tokenVerificationId,
+                  tokenVerified: false
                 })
               }
             )
             .then(
               function (keyFetchToken) {
                 response.keyFetchToken = keyFetchToken.data.toString('hex')
+                response.challengeMethod = 'email';
+                response.challengeReason = 'signin';
```

Comment:
I've applied the following diff locally to make both of the above conditions so, though I'm sure there is a cleaner way of doing it:

```diff
diff --git a/lib/routes/account.js b/lib/routes/account.js
index 0c5b613..4475a6f 100644
--- a/lib/routes/account.js
+++ b/lib/routes/account.js
@@ -513,7 +513,9 @@ module.exports = function (
               function (keyFetchToken) {
                 response.keyFetchToken = keyFetchToken.data.toString('hex')
                 response.challengeMethod = 'email';
-                response.challengeReason = 'signin';
+                response.challengeReason = emailRecord.emailVerified ? 'signin' : 'signup';
+                response.verified = false;
+
                 return response
               }
             )
```